radon/tests/test_raw_visitor.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+from radon.raw import Module, analyze
+from radon.raw_visitor import RawVisitor
+from radon.tests import test_raw
+
+# only testing cases with functions, and remove test with trailing
+# comment since this is not in the function scope.
+reuseable_tests = test_raw.ANALYZE_CASES[9:13] + test_raw.ANALYZE_CASES[14:]
+@pytest.mark.parametrize('code, expected', reuseable_tests)
+def test_raw_visitor_functions(code, expected):
+    code = test_raw.dedent(code)
+    raw_visitor = RawVisitor.from_code(code)
+    # only one function in these tests
+    raw_result = raw_visitor.functions[0]
+    # exclude the details about function name, lineno, etc. for now
+    formated_result = Module(*raw_result[7:])
+    assert formated_result == Module(*expected), '\n result: \
+        {}\n expected: {}'.format(formated_result, Module(*expected))
+    assert formated_result.loc == formated_result.blank \
+                                    + formated_result.sloc \
+                                    + formated_result.single_comments \
+                                    + formated_result.multi
+
+# @pytest.mark.parametrize('code,expected', ANALYZE_CASES)
+# def test_analyze(code, expected):
+#     code = dedent(code)
+
+#     try:
+#         len(expected)
+#     except:
+#         with pytest.raises(expected):
+#             analyze(code)
+#     else:
+#         result = analyze(code)
+#         assert result == Module(*expected)
+#         assert result.loc == result.blank + result.sloc + result.single_comments + result.multi
```

Comment:
These tests need to be enabled so that we know which ones don't pass.

Reply:
@rubik , I suggest deleting these tests since they are not relevant for raw_visitor.  The scope of raw.py and raw_visitor.py are different.  For example the first couple of tests in test_raw.ANALYZE_CASES include strings and inline code that is not wrapped in a function which should return zeros when metrics are computed from test_raw_visitor.py, but not from raw.py.

I wanted to reuse the test cases but I wasn't sure what the best approach was so this was just a quick hack to filter out the reuseable ones.  It probably makes sense to separate out the reuseable tests so the current indexing of test_raw.ANALYZE_CASES isn't so fragile as implemented in test_raw_visitor.py.

Reply:
@BlaneG I see. It makes sense, you can go ahead like you say.